### DISCUSSION TITLE
Using a clipping plane

### DISCUSSION THREAD

**Discussion by ShaharKlingerSSYS** [raysan5/raylib#3042] [Q&A] [answered]
Hello everyone,

I want to use a clipping plane to slice my scene, but I can't seem to understand how to implement this with raylib. From what I understand, I need to enable OpenGL's `GL_CLIP_DISTANCE`, but raylib doesn't offer a `glEnable` function.

How do I use low-level OpenGL calls with raylib, or how do I perform fast clipping of an entire scene?

Thanks for helping!

**Comment by raysan5** [comment]
What are you trying to do? Maybe what you need is just a `RenderTexture`?

**Reply by ShaharKlingerSSYS** [reply]
I'm trying to "slice" an entire scene, for example in the Y axis (up), as seen in [this video](https://www.youtube.com/watch?v=0NH9k4zTAqk&ab_channel=ThinMatrix). I'm trying to avoid manually clipping the meshes, but instead leverage OpenGL's built-in functionality.

**Comment by chriscamacho** [comment, marked as answer]
if you do this:

#define GL_GLEXT_PROTOTYPES 1
#include <GL/glcorearb.h>

you can call GL directly, BUT you need to remember that GL is a state machine, so you might need to restore some things so raylib can continue to render...

I used this technique to let Nuklear GUI to render and still work with Raylib
https://www.youtube.com/watch?v=9aGWqa5pL1A
There is template code on my website using Nuklear (bedroomcoders.co.uk)

**Reply by ShaharKlingerSSYS** [reply]
Great, thank you, I'll have a look.
